proposals/0002.md

```diff
@@ -0,0 +1,318 @@
+* Title: Structured error handling in Rust
+* Champion: @CodeSandwich
+* Conversation (optional): [thread on radicle.community](https://radicle.community/t/proposal-structured-error-handling-in-rust/179)
+* References:
+  * [The previous error handling proposal](https://github.com/radicle-dev/radicle-decisions/blob/master/proposals/0001.md)
+  * [`std::error::Error`](https://doc.rust-lang.org/std/error/trait.Error.html)
+  * [A discussion about the good design of errors](https://users.rust-lang.org/t/do-people-not-care-about-printable-error-chains-a-k-a-how-to-nicely-implement-display-for-an-error/35362)
+
+# Summary/Motivation/Introduction
+We need good error handling to make our lives easier.
```

Comment:
This sentence feels like it's not quite there. First, we know that good error handling is a good thing, but what does it mean to make our lives easier? I'd prefer if you would describe what problems we have at the moment and how this proposal will fix or improve it.

Reply:
I'd prefer not to add this detail here. This is an internal technical proposal, not a published book about testing, we already know why error handling is important.

Reply:
I agree with you, as you can see from my second sentence above.